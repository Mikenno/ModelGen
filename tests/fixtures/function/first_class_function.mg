
func vec3(x, y, z)
	v = {"x": x, "y": y, "z": z}

	v.length_squared = func()
		return v.x * v.x + v.y * v.y + v.z * v.z

	v.zyx = func()
		return vec3(v.z, v.y, v.x)

	return v


a = vec3(1, 2, 3)
b = vec3(4, 5, 6)
c = a.zyx()

print(a.x, a.y, a.z)
print(a.length_squared())
print()

print(b.x, b.y, b.z)
print(b.length_squared())
print()

print(c.x, c.y, c.z)
print(c.length_squared())
print()

a.x = b.x
b.x = -b.x

print(a.x, a.y, a.z)
print(a.length_squared())
print()

print(b.x, b.y, b.z)
print(b.length_squared())
print()

print(c.x, c.y, c.z)
print(c.length_squared())
print()
