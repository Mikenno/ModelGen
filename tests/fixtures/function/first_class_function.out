1 2 3
14

4 5 6
77

3 2 1
14

4 2 3
29

-4 5 6
77

3 2 1
14

